src/activity_handlers.cpp

```diff
@@ -4006,6 +4008,9 @@ void activity_handlers::study_spell_do_turn( player_activity *act, Character *yo
         if( studying.get_level() > old_level ) {
             you->add_msg_if_player( m_good, _( "You gained a level in %s!" ), studying.name() );
         }
+} else if ( old_level >= max_level_from_study ) {
+end_activtity_function();
```

Comment:
```suggestion
act->moves_left = 0;
```

Should let the activity end like normal. Wasn't sure exactly how this activity actor worked so I was shooting from the hip before